### PR TITLE
Analysis Redirect

### PR DESCRIPTION
**Pull request by jmthibault79**
Also some new UBam tests for `X-Force-Location` and Describe URLs

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/vault/services/analysis/AnalysisDescribeServiceSpec.scala [hunk 1]
```diff
@@ -31,17 +48,48 @@ class AnalysisDescribeServiceSpec extends VaultFreeSpec with DescribeService {
...
+    }
+
+    "when calling GET to the " + path + " path with a Vault ID" - {
+      "should reference the new files" in {
+        Get(path + "/" + testId) ~> Cookie(HttpCookie("iPlanetDirectoryPro", openAmResponse.tokenId)) ~> describeRoute ~> check {
+          status should equal(OK)
+          val respAnalysis = responseAs[Analysis]
+          respAnalysis.id should equal(testId)
+          respAnalysis.files.get shouldNot be(empty)
+          respAnalysis.files.get.getOrElse("bam", "error") should startWith("http")   // TODO: better way to distinguish URL from UUID?
```
Comment (davidangb) [inline src/test/scala/org/broadinstitute/dsde/vault/services/analysis/AnalysisDescribeServiceSpec.scala]: how about checking to see if the value parses as a UUID?

```
          intercept[IllegalArgumentException] {
            java.util.UUID.fromString(respAnalysis.files.get.getOrElse("bam", "error"))
          }
```

**Comment by davidangb** [comment]
Looking quite good. I have one comment inline about tests. Should be able to merge after that.

This shares SO MUCH code with the ubam redirect service, with a few important differences (ubam.files is a Map, while analysis.files is an Option[Map], for instance). Did you consider trying to use the same code path? I'm definitely not going to hold this review for that - we can refactor this later.

**Comment by jmthibault79** [comment]
Added some nifty test functionality.

Yeah, I agree that this is ripe for refactoring.  It's just my preference to refactor separately from adding functionality.

**Comment by davidangb** [comment]
Love it - nice job on those matchers. Code looks good and passes locally.

The variable name "left" in the custom matchers threw me. Are they using Eithers somewhere in there? The argument is just a string. But I see that's the same variable name used in the BePropertyMatcher example.

I know how much you like to rebase. Back to you to do that, if you want. Feel free to merge yourself or send back to me.

:+1:

**Pull request merged**